Clarify Elasticsearch 9.x compatibility (#2928)
elastic/elasticsearch-py: docs/reference/index.md

## Compatibility [_compatibility]

> Language clients are forward compatible; meaning that the clients support communicating with greater or equal minor versions of {{es}} without breaking. It does not mean that the clients automatically support new features of newer {{es}} versions; it is only possible after a release of a new client version. For example, a 8.12 client version won’t automatically support the new features of the 8.13 version of {{es}}, the 8.13 client version is required for that. {{es}} language clients are only backwards compatible across minor versions with default distributions and without guarantees made. To upgrade to a new major version, you first need to upgrade Elasticsearch, then the Python Elasticsearch client.
> 
> | Elasticsearch version | elasticsearch-py branch |
> | --- | --- |
> | main | main |
> | 9.x | 9.x |
> | 9.x | 8.x |
> | 8.x | 8.x |

Reviewer: ```suggestion
Language clients are _forward compatible:_ each client version works with equivalent or later minor versions of {{es}} without breaking. 

Compatibility does not mean full feature parity. New {{es}} features are supported only in equivalent and later client versions. For example, an 8.12 client fully supports {{es}} 8.12 features and works with 8.13 without breaking; however, it does not support new {{es}} 8.13 features. An 8.13 client fully supports {{es}} 8.13 features.

| Elasticsearch version | elasticsearch-py branch |
| --- | --- |
| main | main |
| 9.x | 9.x |
| 9.x | 8.x |
| 8.x | 8.x |

{{es}} language clients are also _backward compatible_ across minor versions &mdash; with default distributions and without guarantees. 

:::{tip}
To upgrade to a new major version, first upgrade {{es}}, then upgrade the Python {{es}} client.
:::
```

Reviewer: preview (screenshot) of these changes + the edit to the line below 

<img width="744" alt="Screenshot 2025-04-16 at 3 40 16 PM" src="https://github.com/user-attachments/assets/5645f137-3762-4497-a820-29d916cc4f2c" />

> If you have a need to have multiple versions installed at the same time older versions are also released as `elasticsearch7` and `elasticsearch8`.

Reviewer: ```suggestion
If you need to work with multiple client versions, note that older versions are also released as `elasticsearch7` and `elasticsearch8`.
```

Reviewer: preview (screenshot) of these changes:

<img width="744" alt="Screenshot 2025-04-16 at 3 40 16 PM" src="https://github.com/user-attachments/assets/0c661d16-d72a-43d4-8a10-7c4c2835b0eb" />
